### PR TITLE
[BugFix] Prohibit the use of CTEs with the same name in the same level of name scope

### LINKED ISSUE #13047: Same CTEs should not be allowed in a SQL
### Steps to reproduce the behavior (Required)
```sql
with 
cte1 as ( select lo_orderkey, array_agg(lo_orderkey) cnt from lineorder group by lo_orderkey), 
cte1 as (select count(*) from cte1 group by array_length(cnt)) 

select * from cte1 limit 1;

```

### Expected behavior (Required)

The `cte1` is used many times in the query, this should not be allowed.

### StarRocks version (Required)
- main branch

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13047

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 11 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|8|10|80.00%|[123, 130]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 1]
```diff
@@ -118,37 +118,41 @@ private Scope analyzeCTE(QueryRelation stmt, Scope scope) {
...
+                    outputFields.add(new Field(withQuery.getColumnOutputNames().get(fieldIdx), originField.getType(), tableName,
                             originField.getOriginExpression()));
-                    columnOutputNames.add(withQuery.getColumnOutputNames() == null ? originField.getName() :
-                            withQuery.getColumnOutputNames().get(fieldIdx));
                 }
-                withQuery.setColumnOutputNames(columnOutputNames.build());
+
+                /*
+                 *  Because the analysis of CTE is sensitive to order
+                 *  the latter CTE can call the previous resolved CTE,
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:150]: `latter ` -> `later`

**Review by murphyatwork — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 11 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|8|10|80.00%|[123, 130]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|1|1|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**